docker/tools/search_tool.py

```
@@ -0,0 +1,63 @@
+from models.search_response import SearchResponse
+from models.search_result import SearchResult
+from services.search_processor import process_search_results
+from services.search_service import fetch_with_fallback
+
+logger = logging.getLogger(__name__)
+
+# Get API key from environment
+SERPER_API_KEY = os.environ.get("SERPER_API_KEY", "")
+
+
+async def search_tool(query: str, ctx=None) -> dict:
```

Ruff (0.13.2)

Unused function argument: `ctx`

(ARG001)